src/rbac/defaultRoleManager.ts

```diff
@@ -179,20 +181,34 @@ export class DefaultRoleManager implements RoleManager {
     this.domainMatchingFunc = fn;
   }
 
+  /**
+   * addDomainHierarchy sets a rolemanager to define role inheritance between domains
+   * @param rm RoleManager to define domain hierarchy
+   */
+  public async addDomainHierarchy(rm: RoleManager): Promise<void> {
+    this.hasDomainHierarchy = true;
+    this.domainHierarchyManager = rm;
+  }
+
   private generateTempRoles(domain: string): Roles {
     loadOrDefault(this.allDomains, domain, new Roles());
```

Comment:
Hmm. @nodece can you explain what this is for? Just to add empty `Roles` to `domain` if it does not exist?

Reply:
Pretty sure it isn't needed since it's also done in the `extraDomain` loop below.